_health-care/_js/components/HealthCareApp.jsx

```diff
@@ -220,16 +219,39 @@ class HealthCareApp extends React.Component {
   getExternalData(applicationData, statePath) {
     switch (statePath[0]) {
       case 'financial-assessment':
+        if (statePath[1] === 'annual-income') {
```

Comment:
I think I see what you were saying now about calling functions here instead of just passing down data. A way around this is to call this function within `publishStateChange` instead as a callback on `this.setState`. `this.setState` can take a callback of a function to run once the state has been updated (see line 245 of an example of this). So in `publishStateChange` let's do: 
```javascript
this.setState({ applicationData: newApplicationData }, () = {
  this.updateChildrenIncome();
});
```
And then here we can just pass `childrenIncome: this.state.applicationData['financial-assessment']['annual-income'].children`. I think that'll work.